File: Phidget Code/src/Models/Goal.java
package Models;

import com.phidgets.InterfaceKitPhidget;
import com.phidgets.PhidgetException;
import com.phidgets.event.InputChangeEvent;
import com.phidgets.event.InputChangeListener;

/**
 * Created by bri_e on 07-03-17.
 */
public class Goal {


    private double vibrationHappend;
    private double passageHappend;
    private int indexIRSensor;
    private int indexVibrationSensor;
    private int goal;
    private InterfaceKitPhidget interfaceKitPhidget;

    public Goal(InterfaceKitPhidget ifk, int indexPassage, int indexVibration) throws PhidgetException {
        interfaceKitPhidget = ifk;
        indexIRSensor = indexPassage;
        indexVibrationSensor = indexVibration;
        passageHappend = 0;
        vibrationHappend = 0;
        goal = 0;
        ifk.addInputChangeListener(new InputChangeListener() {
            @Override
            public void inputChanged(InputChangeEvent inputChangeEvent) {
                System.out.println("Passage !");
                try {
                    if (ifk.getSensorValue(indexIRSensor) < 100) {
                        passageHappend = System.currentTimeMillis();
                    }
                } catch(PhidgetException e) {
                    System.out.println("Exception à la lecture du phidget de proximité : " + e);
                }
            }
        });
        ifk.addInputChangeListener(new InputChangeListener() {
            @Override
            public void inputChanged(InputChangeEvent inputChangeEvent) {
                System.out.println("Vibration !");
                try {
                    if (ifk.getSensorValue(indexVibration) > 700) {
                        vibrationHappend = System.currentTimeMillis();
                    }
                } catch(PhidgetException e){
                    System.out.println("Exception à la lecture du phidget de vibration : " + e);
                }
                if((vibrationHappend-passageHappend)<10){
                    goal++;
                    System.out.println("Gooaaaaaaaaaaal !!!!!!");
                }
            }
        });
    }

    public int getGoal(){
        return goal;
    }




}

File: PhidgetScala/src/Models/In/Goal.scala
package Models.In

import java.io.FileInputStream

import BlackBox.BlackBox
import Events._
import com.phidgets.InterfaceKitPhidget
import com.phidgets.event.{SensorChangeEvent, SensorChangeListener}
import sun.audio.{AudioPlayer, AudioStream}

/**
  * Created by bri_e on 20-04-17.
  * This object represent a goal, we listen to both passage event and vibration event.
  * With a calculation on the time lapse between both event in a correct order we will determinate if there is a goal or not
  */
class Goal(interfaceKitPhidget: InterfaceKitPhidget, indexPassage: Int, indexVibration: Int, blackBox: BlackBox) {

  // Constants for the goal's sensors.
  private val TIMEOUT_BTW_TWO_GOALS:Long = 8000
  private val INDEX_IR = 3
  private val INDEX_VIB = 4

  // Number of goals and time of the last goal.
  var goal:Int = 0
  var lastGoal:Long = 10000000

  // getters
  def getGoal:Int = goal
  def getLastGoal:Long = lastGoal

  // increments the goal
  def incrementGoal(time:Long):Boolean = {
    if (lastGoal-time < TIMEOUT_BTW_TWO_GOALS) {
      try {
        // open the sound file as a Java input stream
        val gongFile = "/root/Music/goal.wav"
        val in = new FileInputStream(gongFile)

        // create an audiostream from the inputstream
        val audioStream = new AudioStream(in)

        // play the audio clip with the audioplayer class
        AudioPlayer.player.start(audioStream)
      } catch {
        case e:Exception => println("Exception while opening theme" + e)
      }
      println("Gooaaaal !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! the time set :" + time)
      goal += 1
      println("Goal number : " + goal)
      lastGoal = time
      true
    } else {
      println("Too soon")
      false
    }
  }
}
